pkg/rbd/controllerserver.go

```diff
@@ -247,6 +286,18 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 
 	rbdVol := &rbdVolume{}
 	if err := genVolFromVolID(rbdVol, volumeID, cr); err != nil {
+		// If error is ErrInvalidVolID it could be a version 1.0.0 or lower volume, attempt
+		// to process it as such
+		if _, ok := err.(ErrInvalidVolID); ok {
+			if cs.MetadataStore != nil && isLegacyVolumeID(volumeID) {
+				klog.V(2).Infof("attempting deletion of potential legacy volume (%s)", volumeID)
+				return cs.DeleteLegacyVolume(req, cr)
+			}
+
+			// Consider unknown volumeID as a successfully deleted volume
+			return &csi.DeleteVolumeResponse{}, nil
```

Comment:
Seems like if the volume id is an invalid format, it should through an error and if it's a known legacy format and you don't have the metadata store configured, it should be yet another.

Reply:
Ah ok, this is a weirdness of the CSI spec tester that expects that an invalid/unknown VolID is a success (see the test for the same [here](https://github.com/kubernetes-csi/csi-test/blob/65a7181bfd6ab2ed4753816774014bd8b50615ae/pkg/sanity/controller.go#L855-L864)).

Now, we have enough data to ensure the VolID is valid, and can return errors, but the effort to pass the spec test may require us to treat this as a success. Thoughts?

Reply:
Given that test, how about it throws an error if it sees that it's a legacy volid but you haven't configured the metadata store?

Reply:
Oh yes, my bad. Got distracted with the csi-test reference... fixing.

Reply:
fixed (moved the store check to the legacy function to reduce nested if's and really just make the linters happy).